pgpool2-4.6.yaml

```diff
@@ -0,0 +1,255 @@
+    - name: "Configure and start pgpool"
+      runs: |
+        mkdir -p /etc/pgpool2 /var/log/pgpool /tmp/pgpool
+        # Set up PCP authentication
+        echo "${PGHOST}:${PCP_PORT}:${PG_USER}:${PG_PASSWORD}" > $PCPPASSFILE
+        chmod 600 $PCPPASSFILE
+        echo "${PG_USER}:$(pg_md5 $PG_PASSWORD)" > /etc/pgpool2/pcp.conf
+        # Set up pgpool config
+        cat <<EOF > /etc/pgpool2/pgpool.conf
+        listen_addresses = '*'
+        port = ${PGPOOL_PORT}
+        backend_hostname0 = 'localhost'
+        backend_port0 = '${PG_PORT}'
+        backend_weight0 = 1
+        backend_data_directory0 = '${PG_DATA}'
+        enable_pool_hba = off
+        load_balance_mode = off
+        sr_check_user = '${PG_USER}'
+        sr_check_password = '${PG_PASSWORD}'
+        pcp_port = '${PCP_PORT}'
+        pcp_config_file = '/etc/pgpool2/pcp.conf'
+        health_check_period = 2
+        health_check_timeout = 2
+        health_check_user = '${PG_USER}'
+        health_check_password = '${PG_PASSWORD}'
+        EOF
+        # Start pgpool2
+        pgpool -n -f /etc/pgpool2/pgpool.conf &
+        sleep 5
+    - name: "Test query via pgpool port"
+      runs: |
+        # connect via pgpool and validate connection ; retry in case pgpool is not up yet
+        for attempt in $(seq 1 10) ; do
+          if ! psql -U${PG_USER} -h${PGHOST} -p${PG_PORT} ${PG_DB} -c "SELECT * FROM test_table" \
+            | grep 'Test value' ; then
+            sleep 3
+          else
+            exit 0
+          fi
+        done
+        echo "Unable to query data"
+        exit 1
+    - name: "Test pcp_node_count"
+      runs: |
+        pcp_node_count -U "$PG_USER" -h "$PGHOST" -p "${PCP_PORT}" | grep -q "1"
+    - name: "Test healtcheck via pcp_node_info"
+      runs: |
+        pcp_node_info -U "$PG_USER" -h "$PGHOST" -p "$PCP_PORT" -w 0 | grep -q "primary"
+    - name: "Check healthcheck stats"
+      runs: |
+        pcp_health_check_stats -U "$PG_USER" -h "$PGHOST" -p "$PCP_PORT" -w -v 0 | grep -q "primary"
+    - name: "Display and validate Pgpool childprocesses"
+      runs: |
+        pcp_proc_count -U "$PG_USER" -h "$PGHOST" -p "$PCP_PORT"
+        psql -U $PG_USER -h localhost -p $PGPOOL_PORT -d ${PG_DB} -c "SELECT pg_sleep(60);" & sleep 1
+        pcp_proc_info -U "$PG_USER" -h "$PGHOST" -p "$PCP_PORT" --all | grep -q "Execute command"
+    - name: "Check and validate pcp_pool_status"
+      runs: |
+        pcp_pool_status -U "$PG_USER" -h "$PGHOST" -p "$PCP_PORT" | grep -q "wolfi" #used sr_check_user
+    - name: "Detach primary node"
+      runs: |
+        pcp_detach_node -U "$PG_USER" -h "$PGHOST" -p "$PCP_PORT" -n 0 --gracefully | grep -q "Command Successful"
+    - name: "Stop pgpool using pcp_stop_pgpool"
+      runs: |
+        pcp_stop_pgpool -U "$PG_USER" -h "$PGHOST" -p "$PCP_PORT" --mode=fast | grep -q "Command Successful"
+    - uses: test/tw/ldd-check
+      with:
+        packages: ${{package.name}}
```

Comment:
we can consider omitting this because it defaults to using the same.

Reply:
Returns this error in case I omit it:
```
ERRO failed to test package: compiling pgpool2-4.6.yaml tests: compiling "" test pipelines: compiling Pipeline[0]: mutating with: variable context.name not defined
```